src/Compilers/CSharp/Test/Semantic/SourceGeneration/SyntaxAwareGeneratorTests.cs

```diff
@@ -2077,6 +2077,66 @@ void Function()
             Assert.Equal("Simulated cancellation from external source", results.Results[0].Exception!.Message);
         }
 
+        [Fact]
+        public void Syntax_Provider_Doesnt_Attribute_Incorrect_Timing()
+        {
+
+            var source = @"
+class C 
+{
+    int Property { get; set; }
+
+    void Function()
+    {
+        var x = 5;
+        x += 4;
+    }
+}
+";
+            var parseOptions = TestOptions.Regular.WithLanguageVersion(LanguageVersion.Preview);
```

Comment:
>TestOptions.Regular.WithLanguageVersion(LanguageVersion.Preview) [](http://example.com/codeflow?start=31&length=64)

Minor: `TestOptions.RegularPreview`

Reply:
nit: No parse option should be needed. `CreateCompilation` default to preview LangVer.